engine/src/cmd/controls_factory.cpp

```diff
@@ -272,14 +271,18 @@ Control* getControl(std::map<std::string, std::string> attributes) {
     // Font
     if(attributes.count("font")) {
         std::vector<double> font_array = splitAndConvert(attributes["font"], ',');
-        Font font(font_array[0], font_array[1]);
-        c->setFont(font);
+        if (font_array.size() < 2) {
+            VS_LOG(error, "controls_factory getControl(): font_array doesn't have enough elements");
```

Comment:
I don't know why this condition is happening. We will probably want to investigate further. But for now, this keeps the game from crashing at this point.

Reply:
So, this raises a bigger philosophical question of how much crap are we willing to let the engine take from the assets?

For JSON files, the file obviously needs to be formatted correctly or parsing will fail. Most JSON files are crucial to the game, so may as well crash it with an error.

But this is inside content and for a specific control so it points to an error in how I converted fonts to controls. Best open an issue/bug for this.

Reply:
Filed as https://github.com/vegastrike/Assets-Masters/issues/56